src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -118,13 +119,22 @@ public Response createDatasetRegistration(
           registration,
           user,
           files);
-      Study study = datasets.get(0).getStudy();
+      Integer studyId = datasets.get(0).getStudyId();
+      Study study = datasetService.findStudy(studyId);
       DatasetRegistrationSchemaV1Builder builder = new DatasetRegistrationSchemaV1Builder();
       DatasetRegistrationSchemaV1 createdRegistration = builder.build(study, datasets);
-      URI uri = UriBuilder.fromPath(String.format("/api/dataset/study/%s", study.getStudyId()))
-          .build();
-      String entity = GsonUtil.buildGsonNullSerializer().toJson(createdRegistration);
-      return Response.created(uri).entity(entity).build();
+      if (study != null) {
+        URI uri = UriBuilder.fromPath(String.format("/api/dataset/study/%s", study.getStudyId()))
+            .build();
+        String entity = GsonUtil.buildGsonNullSerializer().toJson(createdRegistration);
+        return Response.created(uri).entity(entity).build();
+      } else {
+        Exception entityException =
+            new UnprocessableEntityException("Study was not found after it was created");
+        logException(entityException);
+        throw entityException;
+      }
```

Comment:
I think you could invert this if_else block to simplify this 
```suggestion
      if (study == null) {
          Exception entityException = new UnprocessableEntityException("Study was not found after it was created");
          logException(entityException);
          throw entityException;
      } 
      
      DatasetRegistrationSchemaV1Builder builder = new DatasetRegistrationSchemaV1Builder();
      DatasetRegistrationSchemaV1 createdRegistration = builder.build(study, datasets);
      String entity = GsonUtil.buildGsonNullSerializer().toJson(createdRegistration);
      URI uri = UriBuilder.fromPath(String.format("/api/dataset/study/%s", study.getStudyId())).build();
     
      return Response.created(uri).entity(entity).build();
```

Reply:
Good eye!  Thank you.